### ISSUE TITLE
COALESCE issues with null and timestamp as first parameters

### ISSUE THREAD

**Issue by aviv-vonage** [oguimbal/pg-mem#178]
# Describe the bug

I noticed two different cases where COALESCE is not working properly. 
The first one is when passing null as the first property in the COALESCE query. 
The second case is when passing Timestamp as the first property in the COALESCE query. 
I saw that Pg-mem coalesce function is trying to convert all the arguments to the first argument's type (why is it necessary?). 
Here is the stack for the second case:

```
Error: cannot cast type timestamp to text

🐜 This seems to be an execution error, which means that your request syntax seems okay,
but the resulting statement cannot be executed → Probably not a pg-mem error.

*️⃣ Failed SQL statement: UPDATE x SET expiration_date=COALESCE('2021-12-07T13:49:53.458Z', expiration_date), expire_after_use=COALESCE(true, expire_after_use), theme_id=COALESCE(null, theme_id) WHERE name='name' RETURNING *;;

👉 You can file an issue at https://github.com/oguimbal/pg-mem along with a way to reproduce this error (if you can), and  the stacktrace:


    at /node_modules/pg-mem/src/datatypes/datatype-base.ts:160:23
    at TimestampType._convert (/node_modules/pg-mem/src/datatypes/datatype-base.ts:192:25)
    at TimestampType.cast (/node_modules/pg-mem/src/datatypes/datatype-base.ts:158:21)
    at Evaluator.cast (/node_modules/pg-mem/src/evaluator.ts:240:26)
    at /node_modules/pg-mem/src/function-call.ts:49:36
    at Array.map (<anonymous>)
    at Object.buildCall (/node_modules/pg-mem/src/function-call.ts:49:25)
    at Object.function (/node_modules/pg-mem/src/evaluator.ts:347:16)
    at _buildValueReal (/node_modules/pg-mem/src/expression-builder.ts:100:34)
    at _buildValue (/node_modules/pg-mem/src/expression-builder.ts:48:11)
    at Object.buildValue (/node_modules/pg-mem/src/expression-builder.ts:18:17)
    at /node_modules/pg-mem/src/schema.ts:1338:23
    at Array.map (<anonymous>)
    at DbSchema.createSetter (/node_modules/pg-mem/src/schema.ts:1332:28)
    at DbSchema.executeUpdate (/node_modules/pg-mem/src/schema.ts:1317:29)
    at DbSchema.prepareWithable (/node_modules/pg-mem/src/schema.ts:522:29)
    at DbSchema.executeWithable (/node_modules/pg-mem/src/schema.ts:531:25)
    at DbSchema._execOne (/node_modules/pg-mem/src/schema.ts:167:33)
    at /node_modules/pg-mem/src/schema.ts:118:31
    at Object.pushContext (/node_modules/pg-mem/src/utils.ts:387:16)
    at DbSchema.queries (/node_modules/pg-mem/src/schema.ts:115:47)
    at queries.next (<anonymous>)
    at DbSchema.query (/node_modules/pg-mem/src/schema.ts:84:20)
    at MemPg.query (/node_modules/pg-mem/src/adapters.ts:128:76)
```

# To Reproduce

```sql

pass null as the first argument in a query with Coalesce. 

```

# pg-mem version

2.1.9

(nb: the version in your package.json version is often not precise enough... please run "cat ./node_modules/pg-mem/package.json | grep version" to tell which minor version is actually installed)

**Comment by oguimbal** [comment]
Hi, this should be fixed in `pg-mem@2.2.0` (reopen if not !)

Sorry for the delay !

**Comment by aviv-vonage** [comment]
awesome! thanks :)
